pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/SymbolVisitor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.symbols;
+
+/**
+ * Visitor over symbols.
+ */
+public interface SymbolVisitor<R, P> {
```

Comment:
This looks like another API PMD provides now... we'll definitively need good documentations and usage examples...